backend/src/Service/OrderLogs/OrderLogsService.php

```diff
@@ -42,6 +48,52 @@ public function createOrderLogsRequest(OrderEntity $order, StoreOwnerBranchEntit
        if($branch) {
          $request->setStoreOwnerBranch($branch);
        }
+       
        return $this->createOrderLogs($request);
     }
-}
+
+    public function getOrderLogsByOrderId($orderId): ?array
+    {
+      $orderLogs = $this->orderLogsManager->getOrderLogsByOrderId($orderId);
+     
+      $currentStage = $this->orderLogsManager->getCurrentStage($orderId);
+   
+      return $this->getOrderLogsTimeLine($orderLogs, $currentStage);
+    }
+
+    public function getOrderLogsTimeLine(array $orderLogs , array $currentStage): ?array
+    {
+        $response = [];
+        $createDate = "";
+        $orderReceivedDate = "";
+
+        foreach ($orderLogs as $orderLog) {
+            if($orderLog['orderState'] === OrderStateConstant::ORDER_STATE_PENDING) {
+               $createDate = $orderLog['createdAt'];
+            }
+
+            if($orderLog['orderState'] === OrderStateConstant::ORDER_STATE_ON_WAY) {
+               $orderReceivedDate = $orderLog['createdAt'];
```

Comment:
how you calculate orderReceivedDate while its status still ON WAY?

Reply:
When the captain receives the order, the order status is on way
Therefore, this case indicates the beginning of receiving the order